### ISSUE TITLE
Rust error handling: "consider downloading ..." gets lost

### ISSUE THREAD

**Issue by chrysn** [RIOT-OS/RIOT#18200]
#### Description

While rustc produces theoretically correct error messages on missing components, these are of little help to users who just happen to have Rust components (which they didn't write on their own) in their tree (as might happen if after #16833 components were written in Rust).

The issue to decide here (before implementing it) is what the expected behavior is.

#### Steps to reproduce the issue

* Install all relevant tools as described in [the "using Rust" page](https://doc.riot-os.org/using-rust.html)

  If you don't know your CPU architecture off your head (or don't know how the Rust names are built) and copy-paste the installation instructions, only thumbv7m-none-eabi will be installed.

(Alternatively, `apt install cargo` for different errors)

* Build something with Rust on a different architecture, eg. microbit-v2 that uses thumbv7m-none-eabi**hf**.

#### Expected results

One of

1. the build system refusing to build, and tells you to `rustup target add thumbv7em-none-eabihf`
2. the build system just adds the relevant target to the toolchain.

I'm not a big fan of automagically installing stuff, but then again rustup is already installed, and its handling of toolchains and components seems to be geard toward making components addable easily. There is some precedent for loading data system-wide where it's managed somehow (BUILD_IN_DOCKER=1 fetches and persists riotbuild), but that has a means of removing unused stuff (`docker image prune`) which rustup has not.

Thing is: 2 is way easier to implement -- it's just a single command that'd fetch anything missing from the net, and no-op in 0.1s when all is there, and maybe what more users want.

To big question:

***Is 2 the way we want do do these things?***

#### Actual results

```
error[E0463]: can't find crate for `core`
  |
  = note: the `thumbv7em-none-eabihf` target may not be installed
  = help: consider downloading the target with `rustup target add thumbv7em-none-eabihf`
```

followed by a litany of other errors (which is bad on its own, but already tracked at https://github.com/rust-lang/rust/issues/97685)

[edit: There are two more relevant issues being tracked on the Rust side on making these more helpful -- [pointing out the `--toolchain` needed with the `rustup target add`](https://github.com/rust-lang/rust/issues/91600), and [suggesting -Zbuild-std](https://github.com/rust-lang/rust/issues/91459) which we'll probably need on AVR; the difficulty to do 1 is tracked [in rustup](https://github.com/rust-lang/rustup/issues/3007) now].
